solr/core/src/java/org/apache/solr/handler/component/QueryComponent.java

```diff
@@ -910,8 +910,12 @@ protected void createMainQuery(ResponseBuilder rb) {
         additionalAdded = addFL(additionalFL, "score", additionalAdded);
       }
     } else {
-      // reset so that only unique key is requested in shard requests
-      sreq.params.set(CommonParams.FL, rb.req.getSchema().getUniqueKeyField().getName());
+      if (rb.req.getSearcher().enableLazyFieldLoading) {
+        // reset so that only unique key is requested in shard requests
+        sreq.params.set(CommonParams.FL, rb.req.getSchema().getUniqueKeyField().getName());
+      } else {
+        sreq.params.set(CommonParams.FL, "*");
```

Comment:
I don't understand this change. Why should QueryComponent know about whether lazy loading is enabled or not?